process/logs/logs_processor.go

```diff
@@ -1,57 +1,32 @@
 package logs
 
 import (
-	"github.com/ElrondNetwork/covalent-indexer-go"
-	"github.com/ElrondNetwork/covalent-indexer-go/process/utility"
-	"github.com/ElrondNetwork/covalent-indexer-go/schema"
-	"github.com/ElrondNetwork/elrond-go-core/core"
-	"github.com/ElrondNetwork/elrond-go-core/core/check"
-	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/covalent-indexer-go/schemaV2"
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
 )
 
 type logsProcessor struct {
-	pubKeyConverter core.PubkeyConverter
 }
 
 // NewLogsProcessor creates a new instance of logs processor
-func NewLogsProcessor(pubKeyConverter core.PubkeyConverter) (*logsProcessor, error) {
-	if check.IfNil(pubKeyConverter) {
-		return nil, covalent.ErrNilPubKeyConverter
-	}
-
-	return &logsProcessor{
-		pubKeyConverter: pubKeyConverter,
-	}, nil
+func NewLogsProcessor() *logsProcessor {
+	return &logsProcessor{}
 }
 
-// ProcessLogs converts logs data to a specific structure defined by avro schema
-func (lp *logsProcessor) ProcessLogs(logs []*data.LogData) []*schema.Log {
-	allLogs := make([]*schema.Log, 0, len(logs))
-
-	for _, currLog := range logs {
-		processedLog := lp.processLog(currLog)
-		if processedLog != nil {
-			allLogs = append(allLogs, processedLog)
-		}
-	}
-
-	return allLogs
-}
-
-func (lp *logsProcessor) processLog(logData *data.LogData) *schema.Log {
-	if logData == nil || check.IfNil(logData.LogHandler) {
-		return nil
+// ProcessLog converts logs data to a specific structure defined by avro schema
+func (lp *logsProcessor) ProcessLog(log *transaction.ApiLogs) *schemaV2.Log {
+	if log == nil {
+		return schemaV2.NewLog()
 	}
 
-	return &schema.Log{
-		ID:      []byte(logData.TxHash),
-		Address: utility.EncodePubKey(lp.pubKeyConverter, logData.LogHandler.GetAddress()),
-		Events:  lp.processEvents(logData.LogHandler.GetLogEvents()),
+	return &schemaV2.Log{
+		Address: []byte(log.Address),
```

Comment:
why the address is no longer encoded?

Reply:
because we receive it as "erd1..." string from Elrond proxy and we just cast it to byte + send it